python/lsst/daf/persistence/butler.py

```diff
@@ -1337,15 +1340,23 @@ def _locate(self, datasetType, dataId, write):
                 if not write:
                     # If there is a bypass function for this dataset type, we can't test to see if the object
                     # exists in storage, because the bypass function may not actually use the location
-                    # according to the template. Instead, execute the bypass function and include its results
+                    # according to the template.
+                    # If the user provides a bypass_datasetExists_dataType method use that to set
+                    # location.bypass to True/False.
+                    # Otherwise execute the bypass function and include its results
                     # in the bypass attribute of the location. The bypass function may fail for any reason,
                     # the most common case being that a file does not exist. If it raises an exception
                     # indicating such, we ignore the bypass function and proceed as though it does not exist.
-                    if hasattr(location.mapper, "bypass_" + location.datasetType):
+
+                    bypass = None
+                    if checkExistence:
+                        bypass = self._getBypassFunc(location, dataId, "bypass_datasetExists_")
+                    if not bypass:
                         bypass = self._getBypassFunc(location, dataId)
+
+                    if bypass:
                         try:
-                            bypass = bypass()
-                            location.bypass = bypass
+                            location.bypass = bypass()
```

Comment:
So sometimes `location.bypass` is a `bool` indicating mere existence, and sometimes it's a heavy object like `Exposure`? That doesn't feel right. I think the logic needs cleaning up. Maybe just a `bypass() is None`, but then that completely changes the meaning of `location.bypass`. Maybe some thought needs to go into this, or maybe it doesn't given that this is now effectively a dead package?

Reply:
The code passed `location.bypass` around, so it's much simpler to use it as is.  And the code is dead, so I'd rather keep it this way.  Twist my arm if needs be